### PR TITLE
[CPU] changed checks with_cpu_x86...() to mayiuse()

### PR DESCRIPTION
**Pull request by antonvor**

### CODE REVIEW

**Comment by dmitry-gorokhov** [comment]
@IvanNovoselov @maxnick @alexey-varyzgin @EgorDuplensky 
FYI. Since changing mayiuse output (to force specific ISA) is a common debug technic we have to avoid usage of with_cpu_x86_* routines inside the plugin to preserve consistent behavior. 
BTW, it would be good to have some wrapper for mayiuse to avoid spread of DNNL API throughout the plugin.

**Comment by dmitry-gorokhov** [comment]
@antonvor please make CI green

**Review by dmitry-gorokhov — APPROVED** [review]

**Comment by maxnick** [comment]
> @IvanNovoselov @maxnick @alexey-varyzgin @EgorDuplensky FYI. Since changing mayiuse output (to force specific ISA) is a common debug technic we have to avoid usage of with_cpu_x86_* routines inside the plugin to preserve consistent behavior. BTW, it would be good to have some wrapper for mayiuse to avoid spread of DNNL API throughout the plugin.

The wrapper is a good idea actually. Btw, mayiuse is a wrapper too, Xbyak provides cpu singleton object that can be accessed via cpu() call and all the flags can be checked directly, moreover, it is easier to disable some ISA in that level. So, technically we could use the Xbyak cpu object to retrieve all the necessary info about the host CPU.

**Comment by EgorDuplensky** [comment]
> > @IvanNovoselov @maxnick @alexey-varyzgin @EgorDuplensky FYI. Since changing mayiuse output (to force specific ISA) is a common debug technic we have to avoid usage of with_cpu_x86_* routines inside the plugin to preserve consistent behavior. BTW, it would be good to have some wrapper for mayiuse to avoid spread of DNNL API throughout the plugin.
> 
> The wrapper is a good idea actually. Btw, mayiuse is a wrapper too, Xbyak provides cpu class that can be accessed via cpu() call (though its singleton nature implemented in oneDNN) and all the flags can be checked directly, moreover, it is easier to disable some ISA in that level. So, technically we could use the Xbyak cpu object to retrieve all the necessary info about the host CPU, just moving cpu() call into the CPU plugin.

I guess this change is done mostly for DNNL_MAX_CPU_ISA to be working properly and we have to call eventually mayiuse() which behavior is controlled by this env variable.
Would be good to have motivation as part of commit message btw

File: src/plugins/intel_cpu/src/mkldnn_plugin.cpp [hunk 1]
```diff
@@ -731,7 +731,7 @@ Parameter Engine::GetMetric(const std::string& name, const std::map<std::string,
         IE_SET_METRIC_RETURN(AVAILABLE_DEVICES, availableDevices);
     } else if (name == METRIC_KEY(OPTIMIZATION_CAPABILITIES)) {
         std::vector<std::string> capabilities;
-        if (with_cpu_x86_bfloat16())
+        if (dnnl::impl::cpu::x64::mayiuse(dnnl::impl::cpu::x64::avx512_core_bf16))
             capabilities.push_back(METRIC_VALUE(BF16));
         if (hasAVX512())
```
[the remark this answers is not in the document]
Comment (antonvor) [reply src/plugins/intel_cpu/src/mkldnn_plugin.cpp]: replaced with `dnnl::impl::cpu::x64::mayiuse(dnnl::impl::cpu::x64::avx512_common)`

**Review by dmitry-gorokhov — APPROVED** [review]

**Pull request merged**